### PR TITLE
staging/publishing: Set go1.16 version to go1.16.12 and go1.17 to go1.17.5

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind cleanup
/area dependency security

#### What this PR does / why we need it:

staging/publishing: Set go1.16 version to go1.16.11 and go1.17 to go1.17.4
Part of https://github.com/kubernetes/release/issues/2341

/hold to wait for https://github.com/kubernetes/kubernetes/pull/106835 to be merged

/assign @nikhita @dims @justaugustus 
cc: @kubernetes/release-engineering 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -24,17 +24,17 @@ rules:
       branch: release-1.21
       dir: staging/src/k8s.io/code-generator
     name: release-1.21
-    go: 1.16.10
+    go: 1.16.11
```
Comment (justaugustus) [inline staging/publishing/rules.yaml:27]:
```suggestion
    go: 1.16.12
```

Same for all of the go1.16 references below.

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -52,17 +52,17 @@ rules:
...
       dir: staging/src/k8s.io/apimachinery
     name: release-1.22
-    go: 1.16.10
+    go: 1.16.11
   - source:
       branch: release-1.23
       dir: staging/src/k8s.io/apimachinery
     name: release-1.23
-    go: 1.17.3
+    go: 1.17.4
```
Comment (justaugustus) [inline staging/publishing/rules.yaml:65]:
```suggestion
    go: 1.17.5
```

Same for all of the go1.17 references below.

**Review by justaugustus — CHANGES_REQUESTED** [review]
Since https://github.com/kubernetes/release/issues/2349 is in-flight, I'd suggest updating these rules to reflect the new versions.

File: staging/publishing/rules.yaml (hunk 1 above)
Comment (cpanato) [reply staging/publishing/rules.yaml]: done

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (cpanato) [reply staging/publishing/rules.yaml]: done

**Review by xmudrii — APPROVED** [review]
/lgtm

**Comment by justaugustus** [comment]
/hold cancel
https://github.com/kubernetes/kubernetes/pull/106983 is on track to merge.

**Pull request merged**